### PR TITLE
Tenischev Semen 2537

### PR DESCRIPTION
**Pull request by Tenischev**
Full task

### CODE REVIEW

**Comment by thevery** [comment]
Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]

**Comment by Tenischev** [comment]
The build settings indicated unsigned, fixed

**Comment by thevery** [comment]
image size is wrong on nexus 5/full hd (and images are not even equal in size!)

**Comment by thevery** [comment]
crash on rotate:
E/AndroidRuntime( 5460): FATAL EXCEPTION: main
E/AndroidRuntime( 5460): Process: com.example.extratask1, PID: 5460
E/AndroidRuntime( 5460): java.lang.RuntimeException: Unable to start receiver com.example.extratask1.MyReceiver: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
E/AndroidRuntime( 5460):    at android.app.ActivityThread.handleReceiver(ActivityThread.java:2426)
E/AndroidRuntime( 5460):    at android.app.ActivityThread.access$1700(ActivityThread.java:135)
E/AndroidRuntime( 5460):    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1272)
E/AndroidRuntime( 5460):    at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 5460):    at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime( 5460):    at android.app.ActivityThread.main(ActivityThread.java:5017)
E/AndroidRuntime( 5460):    at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 5460):    at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime( 5460):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
E/AndroidRuntime( 5460):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
E/AndroidRuntime( 5460):    at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime( 5460): Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
E/AndroidRuntime( 5460):    at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
E/AndroidRuntime( 5460):    at java.util.ArrayList.get(ArrayList.java:308)
E/AndroidRuntime( 5460):    at com.example.extratask1.MyReceiver.onReceive(MyReceiver.java:36)
E/AndroidRuntime( 5460):    at android.app.ActivityThread.handleReceiver(ActivityThread.java:2419)
E/AndroidRuntime( 5460):    ... 10 more
W/ActivityManager(  762):   Force finishing activity com.example.extratask1/.MyActivity

**Comment by thevery** [comment]
previews are scaled, not cropped

**Comment by thevery** [comment]
second screen image quality is bad

**Comment by thevery** [comment]
app crashes on click after refresh:
E/AndroidRuntime(10087): FATAL EXCEPTION: main
E/AndroidRuntime(10087): Process: com.example.extratask1, PID: 10087
E/AndroidRuntime(10087): android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
E/AndroidRuntime(10087):    at android.app.ContextImpl.startActivity(ContextImpl.java:1034)
E/AndroidRuntime(10087):    at android.app.ContextImpl.startActivity(ContextImpl.java:1021)
E/AndroidRuntime(10087):    at android.content.ContextWrapper.startActivity(ContextWrapper.java:311)
E/AndroidRuntime(10087):    at com.example.extratask1.MyArrayAdapter$6.onClick(MyArrayAdapter.java:110)
E/AndroidRuntime(10087):    at android.view.View.performClick(View.java:4438)
E/AndroidRuntime(10087):    at android.view.View$PerformClick.run(View.java:18422)
E/AndroidRuntime(10087):    at android.os.Handler.handleCallback(Handler.java:733)
E/AndroidRuntime(10087):    at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(10087):    at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime(10087):    at android.app.ActivityThread.main(ActivityThread.java:5017)
E/AndroidRuntime(10087):    at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(10087):    at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(10087):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
E/AndroidRuntime(10087):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
E/AndroidRuntime(10087):    at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(  762):   Force finishing activity com.example.extratask1/.MyActivity

**Comment by thevery** [comment]
toast is nice but you should show it both on start and finish

**Comment by thevery** [comment]
useless titlebar on first and second screen (you should use actionbar on first and put refresh to it)

**Comment by thevery** [comment]
app is ok (20-8=12), code review pending

**Comment by thevery** [comment]
no icon

File: res/layout/itemdisplayer.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/itemdisplayer.xml:3]: useless layout

File: res/layout/landscape.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/landscape.xml:3]: use GridView instead!

File: src/com/example/extratask1/ItemDisplayer.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * Time: 6:20
+ * To change this template use File | Settings | File Templates.
+ */
+public class ItemDisplayer extends Activity {
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.itemdisplayer);
+        Intent intent = getIntent();
+        int id = intent.getIntExtra("id",0);
```
Comment (thevery) [inline src/com/example/extratask1/ItemDisplayer.java:21]: `"id"` should be const

File: src/com/example/extratask1/MyActivity.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public static int numberPic = 20;
+    public static String dirName = "recentImage";
+    private String yandexApiRecent = "http://api-fotki.yandex.ru/api/recent/";
+    public static boolean startRefresh = false;
+    public static Context context ;
+    public static ListView listView;
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if(getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT)
```
Comment (thevery) [inline src/com/example/extratask1/MyActivity.java:31]: use `layout` (or `values`) and `layout-land` (`values-land`) resource folders instead

File: src/com/example/extratask1/MyActivity.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            orientation = false;
+        else
```
Comment (thevery) [inline src/com/example/extratask1/MyActivity.java:33]: you should use {} with multiline if-else

File: src/com/example/extratask1/MyArrayAdapter.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                image1.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        Intent intent = new Intent();
+                        intent.putExtra("id",position * 2);
+                        intent.setClass(context,ItemDisplayer.class);
+                        context.startActivity(intent);
+                    }
+                });
+                image2.setOnClickListener(new View.OnClickListener() {
```
Comment (thevery) [inline src/com/example/extratask1/MyArrayAdapter.java:104]: ton on copy-paste code

File: src/com/example/extratask1/MyAsyncTask.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        private String link;
+        private String name;
+        private String id;
+        private boolean entryOpen;
+        private int kol = 0;
+
+        @Override
+        public void startElement(java.lang.String uri, java.lang.String localName, java.lang.String qName, org.xml.sax.Attributes attributes) throws SAXException {
+            currentElement = qName;
+            if ("entry".equals(qName)){
```
Comment (thevery) [inline src/com/example/extratask1/MyAsyncTask.java:80]: `"entry"` should be const

File: src/com/example/extratask1/MyReceiver.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+ * Created with IntelliJ IDEA.
+ * User: kris13
+ * Date: 19.01.14
+ * Time: 5:52
+ * To change this template use File | Settings | File Templates.
+ */
+public class MyReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent){
+        CharSequence text = "Loading successful";
```
Comment (thevery) [inline src/com/example/extratask1/MyReceiver.java:25]: strings.xml

**Comment by thevery** [comment]
20-8-4=8

**Pull request closed without merging**